packages/next/server/config.ts

```diff
@@ -45,9 +47,14 @@ const experimentalWarning = execOnce(
   }
 )
 
-export function setHttpAgentOptions(
-  options: NextConfigComplete['httpAgentOptions']
-) {
+export function setHttpClientAndAgentOptions(options: NextConfig) {
+  if (semverGte(process.version, '16.8.0')) {
+    ;(global as any).__NEXT_USE_UNDICI = options.experimental?.enableUndici
+  } else {
```

Comment:
```suggestion
  } else if (options.experimental?.enableUndici) {
```
Looks like we only want to warn if the is being attempted to be used